binding/react-native/ios/Rhino.swift

```diff
@@ -9,116 +9,127 @@
+        do {
+            if let rhino = rhinoPool[handle] {
+                var param: [String: Any] = [:]
                 
-        if let rhino = rhinoPool[handle] {            
-
-            var isFinalized:Bool = false
-            pv_rhino_process(rhino, pcm, &isFinalized)            
+                let isFinalized = try rhino.process(pcm: pcm)
+                param["isFinalized"] = isFinalized
+                
+                if isFinalized {
+                    let inference = try rhino.getInference()
+                    param["isUnderstood"] = inference.isUnderstood
                     
-            if !isFinalized {
-                return resolve([
-                    "isFinalized": false
-                ])
-            }            
+                    if inference.isUnderstood {
+                        param["intent"] = inference.intent
+                        param["slots"] = inference.slots
+                    }
+                }
                 
-            var isUnderstood: Bool = false           
-            pv_rhino_is_understood(rhino, &isUnderstood)
-
-            var inference:Dictionary<String,Any?>            
-            if !isUnderstood {                
-                inference = [
-                    "isFinalized": true,
-                    "isUnderstood": false
-                ]
+                resolve(param)
             } else {
-
-                var cIntent: UnsafePointer<Int8>?
-                var numSlots: Int32 = 0
-                var cSlotKeys: UnsafeMutablePointer<UnsafePointer<Int8>?>?
-                var cSlotValues: UnsafeMutablePointer<UnsafePointer<Int8>?>?
-                pv_rhino_get_intent(rhino, &cIntent, &numSlots, &cSlotKeys, &cSlotValues)
-                
-                let intent = String(cString: cIntent!)
-                var slots = [String: String]()
-                for i in 0..<numSlots {
-                    let slot = String(cString: cSlotKeys!.advanced(by: Int(i)).pointee!)
-                    let value = String(cString: cSlotValues!.advanced(by: Int(i)).pointee!)
-                    slots[slot] = value
+                let (code, message) = errorToCodeAndMessage(RhinoError.RhinoInvalidStateError("Invalid handle provided to Rhino 'process'"))
+                reject(code, message, nil)
+            }
+        } catch let error as RhinoError {
+            let (code, message) = errorToCodeAndMessage(error)
+            reject(code, message, nil)
+        } catch {
+            let (code, message) = errorToCodeAndMessage(RhinoError.RhinoError(error.localizedDescription))
+            reject(code, message, nil)
+        }
+    }
+    
+    private func getResourcePath(_ filePath: String) throws -> String {
+        if (!FileManager.default.fileExists(atPath: filePath)) {
+            if let resourcePath = Bundle.main.resourceURL?.appendingPathComponent(filePath).path {
```

Comment:
You probably want to do that `Bundle(for :type(of: self))` instead since I believe this could fail in certain situations